src/pages/guides/javascript/how-to-add-metadata-to-spl-tokens.md

```diff
@@ -0,0 +1,318 @@
+
+```bash
+npm init -y
+```
+
+### Required Packages
+
+Install the required pacakges for this guide.
+
+{% packagesUsed packages=["umi", "umiDefaults" ,"tokenMetadata", "mpl-toolbox"] type="npm" /%}
+
+```bash
+npm i @metaplex-foundation/umi
+```
+
+```bash
+npm i @metaplex-foundation/umi-bundle-defaults
+```
+
+```bash
+npm i @metaplex-foundation/mpl-token-metadata
+```
+
+```bash
+npm i @metaplex-foundation/mpl-toolbox
+```
+
+### Imports and Wrapper Function
+
+We will list the necessary imports and create two wrappers functions,
+
+1. `createSPLToken` _(Optional, if you do not have an SPL token already)_
+
+2. `addMetadata`
+
+```typescript
+import {
+	createMetadataAccountV3,
+	findMetadataPda,
+	mplTokenMetadata,
+} from "@metaplex-foundation/mpl-token-metadata";
+import { createMint, mplToolbox } from "@metaplex-foundation/mpl-toolbox";
+import { generateSigner, signerIdentity, sol } from "@metaplex-foundation/umi";
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import { base58 } from "@metaplex-foundation/umi/serializers";
+
+/// 
+/// instantiate umi 
+///
+
+// Create SPL token wrapper function
+async function createSPLToken() {
+	///
+	///
+	///  SPL token initialization code will go here
+	///
+	///
+}
+
+// run the function
+createSPLToken();
+
+// Add metadata to an existing SPL token wrapper function
+async function addMetadata() {
+	///
+	///
+	///  code will go in here
+	///
+	///
+}
+
+// run the function
+addMetadata();
+```
+
+## Setting up Umi
+
+This example is going to run through setting up Umi with a `generatedSigner()`. If you wish to set up a wallet or signer differently you can check out the [**Connecting to Umi**](/umi/connecting-to-umi) guide.
+
+You can place the Umi instantiation code inside or outside the code blocks, but to reduce code duplication, we will keep it outside.
+
+### Generating a New Wallet
+
+```ts
+const umi = createUmi("https://api.devnet.solana.com")
+	.use(mplTokenMetadata())
+	.use(mplToolbox());
+
+// Generate a new keypair signer.
+const signer = generateSigner(umi);
+
+// Tell umi to use the new signer.
+umi.use(signerIdentity(signer));
+
+// Airdrop 2 SOL to the identity
+// if you end up with a 429 too many requests error, you may have to use
+// the a different rpc other than the free default one supplied.
+await umi.rpc.airdrop(umi.identity.publicKey, sol(2));
+```
+
+### Use an Existing Wallet Stored Locally
+
+```ts
+const umi = createUmi("https://api.devnet.solana.com")
+	.use(mplTokenMetadata())
+	.use(mplToolbox());
+
+// You will need to us fs and navigate the filesystem to
+// load the wallet you wish to use via relative pathing.
+const walletFile = const imageFile = fs.readFileSync('./keypair.json')
+
+// Convert your walletFile onto a keypair.
+let keypair = umi.eddsa.createKeypairFromSecretKey(new Uint8Array(walletFile));
+
+// Load the keypair into umi.
+umi.use(keypairIdentity(umiSigner));
+```
+
+## Creating the Token
```

Comment:
I feel like instead of how to create the token with the toolbox we should instead recommend to use the helper functions as mentioned here https://developers.metaplex.com/token-metadata/mint#create-helpers since it reduces the required code massively.
Instead this guide could focus on how to use createV1